### DISCUSSION TITLE
Exposed passwords reports

### DISCUSSION THREAD

**Discussion by nicolasvila** [dani-garcia/vaultwarden#2526] [Q&A]
Hi,
Just a simple question. How are implemented compromised passwords check in Vauldwarden?
How are sent the password and how safe this functionnality is?
Thank you

**Comment by BlackDex** [comment]
All is done client side, nothing is send to Vaultwarden at all for this.
Also see: https://bitwarden.com/help/reports/#exposed-passwords-report
